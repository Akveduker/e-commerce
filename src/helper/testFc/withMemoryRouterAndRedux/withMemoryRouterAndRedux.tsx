import { configureStore } from "@reduxjs/toolkit"
import { render } from "@testing-library/react"
import { InitialEntry } from "history"
import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import { mockReducersForStore } from "../../mock/mockStore/mockStore"
export const withMemoryRouterAndRedux = (children: React.ReactNode, reducers?: Partial<typeof mockReducersForStore>, initialEntries?: InitialEntry[]) => {
    const mockStore = configureStore({
        reducer: { ...mockReducersForStore, ...reducers }
    });
    return render(
        <MemoryRouter initialEntries={initialEntries}>
            <Provider store={mockStore}>

                {children}
            </Provider>
        </MemoryRouter>
    )
}