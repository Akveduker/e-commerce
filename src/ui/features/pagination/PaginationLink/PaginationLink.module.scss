@import "../../../../styles/_vars.scss";
@import "../../../../styles/_mixins.scss";
.link {
    margin-left: 8px;
    font-size: 12px;
    @include fontRegular;
    border: none;
    padding: 0;
    cursor: pointer;
    &--active {
        color: $C2A;
    }
}
