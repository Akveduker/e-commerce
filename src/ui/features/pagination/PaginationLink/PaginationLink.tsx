import { FC } from 'react';
import { useSearchParams } from 'react-router-dom';
import { setType } from '../../../../slices/filterSlice/filterSlice';
import { useAppDispatch } from '../../../../store/store';
import s from './PaginationLink.module.scss'
interface PaginationLinkProps {
    num: number
}
const PaginationLink: FC<PaginationLinkProps> = ({ num }) => {
    const dispatch = useAppDispatch()
    const [searchParams, setSearchParams] = useSearchParams()
    let page = +(searchParams.get('p') as string)
    const changePage = () => {
        dispatch(setType('page'))
        setSearchParams(searchParams => {
            searchParams.set('p', num.toString())
            return searchParams
        })
    }
    const className = num === page ? `${s['link']} ${s['link--active']}` : s['link']
    return (
        <button key={num} className={className} onClick={changePage} type='button' data-testid={'button'}>
            {num}
        </button>
    );
};

export default PaginationLink;