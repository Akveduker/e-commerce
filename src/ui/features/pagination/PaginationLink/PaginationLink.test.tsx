import { screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { withMemoryRouterAndRedux } from "../../../../helper/testFc/withMemoryRouterAndRedux/withMemoryRouterAndRedux"
import PaginationLink from "./PaginationLink"

describe('PaginationLink', () => {
    test('Ссылка неактивна', () => {
        const { container } = withMemoryRouterAndRedux(<PaginationLink num={5} />, {}, ['/?p=2'])
        const button = screen.getByTestId('button')
        expect(button).not.toHaveClass('link--active')
        expect(container).toMatchSnapshot()

    })
    test('Ссылка активна,меняет параметры и состояние', () => {
        const { container } = withMemoryRouterAndRedux(<PaginationLink num={1} />, {}, ['/?p=2'])
        const button = screen.getByTestId('button')
        userEvent.click(button)
        expect(button).toHaveClass('link--active')
        expect(container).toMatchSnapshot()
    })
})