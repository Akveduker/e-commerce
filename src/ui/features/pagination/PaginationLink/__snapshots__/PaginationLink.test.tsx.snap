// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationLink Ссылка активна,меняет параметры и состояние 1`] = `
<div>
  <button
    class="link link--active"
    data-testid="button"
    type="button"
  >
    1
  </button>
</div>
`;

exports[`PaginationLink Ссылка неактивна 1`] = `
<div>
  <button
    class="link"
    data-testid="button"
    type="button"
  >
    5
  </button>
</div>
`;
